### DISCUSSION TITLE
why does the mini-css-extract-plugin use ChildCompiler but not Compiler?

### DISCUSSION THREAD

**Discussion by BUPTlhuanyu** [webpack/webpack#14677] [Q&A]
https://github.com/webpack-contrib/mini-css-extract-plugin/blob/0c286f35de463374b61f1936fdf5cc38570bba22/src/loader.js?_pjax=%23repo-content-pjax-container#L245

why does the mini-css-extract-plugin use ChildCompiler but not Compiler? like this? 

```
new Compiler(context)
```

**Comment by alexander-akait** [comment]
Because we need context of main compilation/compiler

**Reply by BUPTlhuanyu** [reply]
@alexander-akait 
Thank you for your reply! 
But when i use thread-loader with mini-css-extract-plugin, the error show that the createChildCompiler is undefined.

**Reply by alexander-akait** [reply]
Do not use `thread-loader`, please prefer to use `cache`, `thread-loader` doesn't decrease time very well

**Reply by BUPTlhuanyu** [reply]
All right! I will try other ways to decrease compile time!
But I hope it can be improved.
